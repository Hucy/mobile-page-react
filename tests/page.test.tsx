import 'jest-styled-components';
import React from 'react';

import { mount, shallow } from 'enzyme';

import Page, { GlobalStyle, PageStyle } from '../src';

test('should render correctly', () => {
  const wrapper = mount(<Page />);
  expect(wrapper).toMatchSnapshot();
});

test('page child should be exist', () => {
  const wrapper = shallow(<Page />);
  expect(wrapper.children().length).toBe(2);
  expect(wrapper.find(PageStyle).exists()).toBe(true);
  expect(wrapper.find(GlobalStyle).exists()).toBe(true);
});
test('custom className should succeed', () => {
  const wrapper = shallow(<Page className="custom-class-name" />);
  expect(wrapper.find(PageStyle).hasClass('custom-class-name')).toBe(true);
});

test('page child should not be null', () => {
  const wrapper = shallow(<Page>ll</Page>);
  expect(wrapper.find(PageStyle).children().length).toBe(1);
});
