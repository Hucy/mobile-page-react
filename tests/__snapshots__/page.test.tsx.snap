// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.c0 {
  height: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  position: absolute;
  width: 100%;
}

<Page>
  <GlobalStyleComponent />
  <Styled(styled.div)>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  height: 100%;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: stretch;
",
              "
  position: absolute;
  width: 100%;
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      />
    </StyledComponent>
  </Styled(styled.div)>
</Page>
`;
